worker/lease/manager.go

```diff
@@ -424,6 +448,14 @@ func (manager *Manager) startRetry() *retry.Attempt {
 	)
 }
 
+func (manager *Manager) handlePin(p pin) {
+	p.respond(errors.Trace(manager.config.Store.PinLease(p.leaseKey)))
```

Comment:
These and similar calls violate the (Law of Demeter)[https://en.wikipedia.org/wiki/Law_of_Demeter], but I tried to limit how much I bit off with this patch.

Reply:
Maaaaaaaaybe? I don't think the config struct here should be considered to be an independent object from a Demeter point of view. The way we use them is just for structuring and brevity - they almost never have behaviour other than validation. It's not clear that the code would be better if the New func assigned those fields one-by-one and the Demeter violation was replaced by `manager.store.PinLease`.

Reply:
I guess it depends how you want to fix it. :)